### PR TITLE
feat(tagging): track pending query

### PR DESCRIPTION
**Pull request by javieri-empathy**
EX-4963

Forces the query tracking to happen immediately in these situations:

- When a search result is clicked.
- When a promoted is clicked.
- When a banner is clicked.
- When a redirection is triggered.
- When a the user triggers the infinite scroll.

This is needed to ensure that we properly track each search request upon any user interaction with the current search response. Otherwise we might lose certain events, as the user might leave the SERP (like on click events), or load a new page (with the infinite scroll).

### CODE REVIEW

File: packages/x-components/src/x-modules/tagging/wiring.ts [hunk 1]
```diff
@@ -81,7 +81,16 @@ export const setSessionDuration = wireCommit('setSessionDuration');
...
+  {
+    cancelOn: 'UserClearedQuery',
+    forceOn: [
+      'UserClickedAResult',
+      'UserClickedAPromoted',
+      'UserClickedABanner',
+      'UserClickedARedirection',
+      'UserReachedResultsListEnd'
+    ]
+  }
```
Comment (javieri-empathy) [inline packages/x-components/src/x-modules/tagging/wiring.ts:93]: This is the actual change, the rest are test related changes. I think I've added any event that might override the query tagging while still consider the search response as accepted.

File: packages/x-components/src/x-modules/search/components/banner.vue [hunk 2]
```diff
@@ -30,6 +30,14 @@
      */
     @Prop({ required: true })
     public banner!: BannerModel;
+
+    /**
+     * Emits the promoted click event.
```
Comment (tiborux) [inline packages/x-components/src/x-modules/search/components/banner.vue:35]:
```suggestion
     * Emits the banner click event.
```

**Review by tiborux — CHANGES_REQUESTED** [review]
GJ, only a little change in the doc

File: packages/x-components/src/adapter/mocked-adapter.ts [hunk 3]
```diff
@@ -71,7 +72,10 @@ class E2ETestsAdapter extends EmpathyAdapter {
   }
 
   track(request: TrackingRequest): Promise<void> {
-    return mockFetch(request, 'track');
+    return fetch(request.url, {
+      method: 'POST',
+      body: JSON.stringify(request.params)
+    }).then(noOp);
```
Comment (tajespasarela) [inline packages/x-components/src/adapter/mocked-adapter.ts:78]: why the `.then(noOp)`?

**Review by tajespasarela — CHANGES_REQUESTED** [review]

File: packages/x-components/src/adapter/mocked-adapter.ts (hunk 3 above)
Comment (javieri-empathy) [reply packages/x-components/src/adapter/mocked-adapter.ts:78]: `SearchAdapter` specifies that `track` method should return `Promise<void>`, but `fetch` returns a `Promise<Response>`. With the `noOp` we are simply ignoring that `Response` and "transforming" it into `void`

**Review by tajespasarela — APPROVED** [review]

**Review by tiborux — APPROVED** [review]

**Pull request merged**